cmd/apprepository-controller/server/controller.go

```diff
@@ -417,10 +420,77 @@ func ownerReferencesForAppRepo(apprepo *apprepov1alpha1.AppRepository, childName
 	return nil
 }
 
+// intervalToCron transforms string durations like "1m" or "1h" to cron expressions
+// Even if valid time units are "ns", "us", "ms", "s", "m", "h",
+// the result will get rounded up to minutes.
+// for durations over 24h only durations below 1 year are supported
+func intervalToCron(duration string) (string, error) {
+	if duration == "" {
+		return "", fmt.Errorf("duration cannot be empty")
+	}
+
+	d, err := time.ParseDuration(duration)
+	if err != nil {
+		return "", fmt.Errorf("error while parsing the duration: %s", err)
+	}
+	cronMins := math.Ceil(d.Minutes()) // round up to nearest minute
+
+	if cronMins < 60 {
+		// https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax
+		// minute(0-59) hour(0-23) dayOfMonth(1-31) month(1-12) dayOfWeek(0-6)
+		return fmt.Sprintf("*/%v * * * *", cronMins), nil // every cronMins minutes
+	}
+
+	cronHours := math.Ceil(d.Hours()) // round up to nearest hour
+	if cronHours < 24 {
+		return fmt.Sprintf("0 */%v * * *", cronHours), nil // every cronHours hours
+	}
+
+	cronDays := math.Ceil(cronHours / 24) // get the days
+	if cronDays < 32 {
+		return fmt.Sprintf("0 0 */%v * *", cronDays), nil // every cronHoursDays days
+	}
+
+	cronMonths := math.Ceil(cronDays / 31) // get the months
+	if cronMonths < 13 {
+		return fmt.Sprintf("0 0 1 */%v *", cronMonths), nil // every cronHoursMonths months
+	}
```

Comment:
New logic for roughly estimating the cron line from a duration

Reply:
Nice - thanks for the update. Just note, small typo in the inline comments on lines 451 and 456 (s/Hours//)